app/controllers/products_controller.rb

```diff
@@ -1,30 +1,24 @@
 class ProductsController < ApplicationController
-  before_action :set_product, only: %i[ show edit update destroy ]
+  before_action :set_product, only: %i[ show edit update destroy]
   before_action :authenticate_user!, only: %i[new edit update destroy] 
+  
   def index
-    @products = Product.paginate(page: params[:page], per_page: 15)
```

Comment:
use a method or constant variable for per_page value 
e.g PER_PAGE = 15
so, it would be easy to change later on